src/H5FDint.c

```diff
@@ -260,148 +300,2075 @@ H5FD_write(H5FD_t *file, H5FD_mem_t type, haddr_t addr, size_t size, const void
 
 /*-------------------------------------------------------------------------
- * Function:    H5FD_get_eof
+ * Function:    H5FD__read_selection_translate
  *
- * Purpose:     Private version of H5FDget_eof()
+ * Purpose:     Translates a selection read call to a vector read call if
+ *              vector reads are supported, or a series of scalar read
+ *              calls otherwise.
  *
- *              This function returns the EOF as a RELATIVE address, i.e.
- *              relative to the base address.  This will be different
- *              from  the end of the physical file if there is a user
- *              block.
+ * Return:      Success:    SUCCEED
+ *                          All reads have completed successfully, and
+ *                          the results havce been into the supplied
+ *                          buffers.
  *
- * Return:      Success:    The EOF address.
+ *              Failure:    FAIL
+ *                          The contents of supplied buffers are undefined.
  *
- *              Failure:    HADDR_UNDEF
+ * Programmer:  NAF -- 5/13/21
+ *
+ * Changes:     None
  *
  *-------------------------------------------------------------------------
  */
-haddr_t
-H5FD_get_eof(const H5FD_t *file, H5FD_mem_t type)
+static herr_t
+H5FD__read_selection_translate(H5FD_t *file, H5FD_mem_t type, hid_t dxpl_id, uint32_t count,
+                               H5S_t **mem_spaces, H5S_t **file_spaces, haddr_t offsets[],
+                               size_t element_sizes[], void *bufs[] /* out */)
 {
-    haddr_t ret_value = HADDR_UNDEF; /* Return value */
-
-    FUNC_ENTER_NOAPI(HADDR_UNDEF)
+    hbool_t         extend_sizes = FALSE;
+    hbool_t         extend_bufs  = FALSE;
+    uint32_t        i;
+    size_t          element_size;
+    void *          buf;
+    hbool_t         use_vector = FALSE;
+    haddr_t         addrs_static[8];
+    haddr_t *       addrs = addrs_static;
+    size_t          sizes_static[8];
+    size_t *        sizes = sizes_static;
+    void *          vec_bufs_static[8];
+    void **         vec_bufs = vec_bufs_static;
+    hsize_t         file_off[H5FD_SEQ_LIST_LEN];
+    size_t          file_len[H5FD_SEQ_LIST_LEN];
+    hsize_t         mem_off[H5FD_SEQ_LIST_LEN];
+    size_t          mem_len[H5FD_SEQ_LIST_LEN];
+    size_t          file_seq_i;
+    size_t          mem_seq_i;
+    size_t          file_nseq;
+    size_t          mem_nseq;
+    size_t          io_len;
+    size_t          nelmts;
+    hssize_t        hss_nelmts;
+    size_t          seq_nelem;
+    H5S_sel_iter_t *file_iter      = NULL;
+    H5S_sel_iter_t *mem_iter       = NULL;
+    hbool_t         file_iter_init = FALSE;
+    hbool_t         mem_iter_init  = FALSE;
+    H5FD_mem_t      types[2]       = {type, H5FD_MEM_NOLIST};
+    size_t          vec_arr_nalloc = sizeof(addrs_static) / sizeof(addrs_static[0]);
+    size_t          vec_arr_nused  = 0;
+    herr_t          ret_value      = SUCCEED;
```

Comment:
In about 30 lines of variable declarations I see the same types (size_t, hbool_t) over and over again. Please gather same-type variables onto the same line to save the poor reader some scrolling.

Reply:
I disagree because of past history dealing with long single line declarations that had caused issues for one reason or another. I can see grouping very similar named vars. But especially with assignments for defaults, single name declarations are more clear. We are not trying to save paper and scrolling is an easy and quick mouse/arrow operation.

Reply:
We've long avoided grouping declarations like that